libs/layout/src/lib/services/media-query/mediaquery.service.ts

```
@@ -0,0 +1,103 @@
+	 */
+	private readonly mediaQueryListenerMap: Map<
+		string,
+		(queryChangedEvent: MediaQueryListEvent) => void
+	> = new Map();
+
+	/**
+	 * A subject that emits the id of the media query that has changed.
+	 */
+	private readonly queryChangedSubject: ReplaySubject<string> = new ReplaySubject();
+
+	constructor(private readonly windowService: WindowService) {}
+
+	/**
+	 * Register a list of media queries that need to be tracked by the service.
+	 *
+	 * @param queries - A list of media queries that should be registered with the service.
+	 */
+	public registerMediaQueries(...queries: [id: string, query: string][]): void {
+		this.windowService.runInBrowser(({ browserWindow }) => {
+			for (let [id, query] of queries) {
+				// Wouter: For each query, create a MediaQueryList object
+				const matchedQuery = browserWindow.matchMedia(query);
+
+				// Wouter: Save the query
+				this.mediaQueryMap.set(id, matchedQuery);
+
+				// Wouter: Emit the id of the query that has changed
+				this.queryChangedSubject.next(id);
+
+				// Wouter: Create a listener for the query. This is done separately to be
+				// able to remove the listener when the service is destroyed
+				const listener = (queryChangedEvent: MediaQueryListEvent) => {
+					this.mediaQueryMap.set(id, queryChangedEvent.currentTarget as MediaQueryList);
+
+					// Wouter: Emit the id of the query that has changed
+					this.queryChangedSubject.next(id);
+				};
+
+				// Wouter: Register the listener to the query
+				matchedQuery.addEventListener('change', listener);
+
+				// Wouter: Save the listener
+				this.mediaQueryListenerMap.set(id, listener);
+			}
+		});
+	}
+
+	/**
+	 * Pass the id of the query whose changes need to be listened to.
+	 *
+	 * @param id - The id of the media query that should be checked.
+	 * @returns An observable that emits a boolean value whenever the requested media query changes.
+	 */
+	public getMatchingQuery$(id: string): Observable<boolean> {
```

I would throw an error here if no query with the id was registered.